arctic/arctic.py

```diff
@@ -183,13 +185,16 @@ def __getstate__(self):
     def __setstate__(self, state):
         return Arctic.__init__(self, **state)
 
-    @mongo_retry
     def list_libraries(self):
         """
         Returns
         -------
         list of Arctic library names
         """
+        return self._list_libraries_cached() if ENABLE_CACHE else self._list_libraries()
```

Comment:
Should we check for authorisation errors? This assumes we have `read`, and sometimes `dbOwner` rights to `meta_db`.